app/settings/surveys/attribute-editor.directive.js

```diff
@@ -85,9 +85,178 @@ function (
+            };
+
+            $scope.displayVideoLabel = function () {
+                return $scope.editAttribute.input === 'video';
+            };
+
+            $scope.displayDateLabel = function () {
+                return $scope.editAttribute.input === 'date';
+            };
+
+            $scope.displayDateTimeLabel = function () {
+                return $scope.editAttribute.input === 'datetime';
+            };
+
+            $scope.displayIntLabel = function () {
+                return $scope.editAttribute.type === 'int';
+            };
+
+            $scope.displayDecimalLabel = function () {
+                return $scope.editAttribute.type === 'decimal';
+            };
+
+            $scope.displayDefaultLabel = function () {
+                return $scope.editAttribute.input !== 'video' && $scope.editAttribute.type !== 'int' && $scope.editAttribute.type !== 'decimal' && $scope.editAttribute.input !== 'date' && $scope.editAttribute.input !== 'datetime' && $scope.editAttribute.input !== 'location' && $scope.editAttribute.input !== 'relation';
+            };
+
+            $scope.errorMsgs = {
+                'num': 'Number(s) only',
+                'video': 'Enter valid video link',
+                'date': 'Enter valid date format',
+                'dateTime': 'Enter valid date & time format',
+                'others': 'hidden'
+            };
+
+            $scope.inputType = [
+                $scope.editAttribute.input === 'number',
+                $scope.editAttribute.input === 'video',
+                $scope.editAttribute.input === 'date',
+                $scope.editAttribute.input === 'datetime'
+            ]
+
+            $scope.errorMessage = function () {
+
+                if ($scope.inputType[0]) {
+                    $scope.errorMsg = $scope.errorMsgs.num;
+                }
+
+                if ($scope.inputType[1]) {
+                    $scope.errorMsg = $scope.errorMsgs.video;
+                }
+
+                if ($scope.inputType[2]) {
+                    $scope.errorMsg = $scope.errorMsgs.date;
+                }
+
+                if ($scope.inputType[3]) {
+                    $scope.errorMsg = $scope.errorMsgs.dateTime;
+                }
+
+                if (!$scope.errorMsg) {
+                    $scope.errorMsg = $scope.errorMsgs.others;
+                }
+
+                return $scope.errorMsg;
+            };
+
+            $scope.canDisplayError = function () {
+
+                let length = $scope.editAttribute.default.length;
+
+                if ($scope.inputType[0] && isNaN($scope.editAttribute.default) && length >= 1) {
+                    return angular.element(document.querySelector('#displayError')).removeClass('hidden') && angular.element(document.querySelector('#form-field')).addClass('error');
```

Comment:
Here, you don't need to write your own validation, check out how input-fields are validated and errors displayed in other parts of the platform, some examples: https://github.com/ushahidi/platform-client/blob/develop/app/settings/surveys/attribute-editor.html#L11
https://github.com/ushahidi/platform-client/blob/develop/app/main/posts/modify/post-value-edit.html#L24
https://github.com/ushahidi/platform-client/blob/develop/app/main/posts/modify/post-value-edit.html#L24
https://github.com/ushahidi/platform-client/blob/develop/app/main/posts/modify/post-value-edit.html#L53

Reply:
Ok @Angamanga I will check out the examples you have sent. Thank you.